cpp/include/cudf/ast/operators.hpp

```diff
@@ -0,0 +1,1159 @@
+/*
+ * Copyright (c) 2020, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include <cmath>
+#include <cudf/types.hpp>
+#include <cudf/utilities/error.hpp>
+#include <cudf/utilities/type_dispatcher.hpp>
+#include <simt/type_traits>
+#include <type_traits>
+#include <utility>
+#include <vector>
+
+namespace cudf {
+
+namespace ast {
+
+/**
+ * @brief Enum of supported operators.
+ *
+ */
+enum class ast_operator {
```

Comment:
We're already in the `ast` namespace, so I'd think this would be redundant. 
```suggestion
enum class operator {
```

Reply:
@jrhemstad `operator` is a [C++ keyword](https://en.cppreference.com/w/cpp/keyword/operator) so I avoided that.

Reply:
ah, yeah, okay.